plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -183,13 +185,7 @@ void deleteModelChunks(String modelId, DeleteResponse deleteResponse, ActionList
             if ((r.getBulkFailures() == null || r.getBulkFailures().size() == 0)
                 && (r.getSearchFailures() == null || r.getSearchFailures().size() == 0)) {
                 log.debug("All model chunks are deleted for model {}", modelId);
-                if (deleteResponse != null) {
-                    // If model metaData not found and deleteResponse is null, do not return here.
-                    // ResourceNotFound is returned to notify that this model was deleted.
-                    // This is a walk around to avoid cleaning up model leftovers. Will revisit if
-                    // necessary.
-                    actionListener.onResponse(deleteResponse);
-                }
+                actionListener.onResponse(deleteResponse);
```

Comment:
So the long pending is caused by the `deleteResponse` is null case ? Curious when the `deleteResponse` will be null?